libs/authorizenet/src/effects/authorize-net.effects.ts

```diff
@@ -66,16 +73,33 @@ export class DaffAuthorizeNetEffects<T extends DaffAuthorizeNetTokenRequest = Da
 		map((actions: Actions[]) => new DaffAuthorizeNetUpdatePaymentFailure('The payment method has failed to update the cart.'))
 	)
 	
-	@Effect({ dispatch: false })
+	@Effect()
   loadAcceptJs$ : Observable<any> = this.actions$.pipe(
     ofType(DaffAuthorizeNetActionTypes.LoadAcceptJsAction),
-		tap((action: DaffLoadAcceptJs) => {
+		map((action: DaffLoadAcceptJs) => {
 			const acceptJsScript = document.createElement('script');
 			acceptJsScript.async = true;
 			acceptJsScript.setAttribute('type', 'text/javascript');
 			acceptJsScript.setAttribute('src', this.config.production ? ACCEPT_JS_PRODUCTION_URL : ACCEPT_JS_SANDBOX_URL);
 			acceptJsScript.setAttribute('charset', 'utf-8');
 			document.body.appendChild(acceptJsScript);
+			return new DaffCheckAcceptJs();
+		})
+	)
+	
+	@Effect()
+	checkAcceptJs$ : Observable<any> = this.actions$.pipe(
+		ofType(DaffAuthorizeNetActionTypes.CheckAcceptJsAction),
+		debounce(() => interval(this.checkAcceptJsRetryTimer)),
+		map(() => {
```

Comment:
Taken from the [Angular docs:](https://angular.io/guide/practical-observable-usage#exponential-backoff)

```bash
function backoff(maxTries, ms) {
 return pipe(
   retryWhen(attempts => zip(range(1, maxTries), attempts)
     .pipe(
       map(([i]) => i * i),
       mergeMap(i =>  timer(i * ms))
     )
   )
 );
}
```

This way, there's no class prop.

This should also fail.

Reply:
Try as I might (for hours), I couldn't get this to work. I converted the Accept check into an observable that would throw an error when it failed, but backoff just let everything through and the check never gets triggered again.

Reply:
added backoff operator